src/test/scala-2.11/markets/performance/MatchingEngineMicroBench.scala

```diff
@@ -0,0 +1,126 @@
+
+/** Base trait for regression testing a MatchingEngine. */
+trait MatchingEngineMicroBench extends Bench.OnlineRegressionReport
+  with MarketsTestKit {
+
+  /** MatchingEngine to use when running the benchmark tests. */
+  def matchingEngine: MatchingEngine
+
+  /** TestKit used to generate testing actors. */
+  def testKit: TestKit
+
+  val config = ConfigFactory.load("regression.conf")
+
+  val askOrderProbability = config.getDouble("matching-engines.input-data.askOrderProbability")
+
+  val limitOrderProbability = config.getDouble("matching-engines.input-data.limitOrderProbability")
+
+  val prng = new Random(config.getLong("matching-engines.input-data.seed"))
+
+  val tradable = TestTradable(config.getString("matching-engines.input-data.symbol"))
+
+  /** Generates a random limit order for some tradable.
+    *
+    * @param tradable
+    * @return
+    */
+  protected def generateLimitOrder(tradable: Tradable): Order = {
+    val sender = testKit.testActor
+    val price = randomLimitPrice(prng)
+    val quantity = randomQuantity(prng)
+    if (askOrderProbability <= prng.nextDouble()) {
+      LimitAskOrder(sender, price, quantity, timestamp(), tradable, uuid())
+    } else {
+      LimitBidOrder(sender, price, quantity, timestamp(), tradable, uuid())
+    }
+  }
+
+  /** Generates a collection of random limit orders for some tradable.
+    *
+    * @param numberOrders
+    * @param tradable
+    * @return
+    */
+  protected def generateLimitOrders(numberOrders: Int, tradable: Tradable) = {
+    for (i <- 1 to numberOrders) yield generateLimitOrder(tradable)
+  }
+
+  /** Generates a random market order for some tradable.
+    *
+    * @param tradable
+    * @return
+    */
+  protected def generateMarketOrder(tradable: Tradable): Order = {
+    val sender = testKit.testActor
+    val quantity = randomQuantity(prng)
+    if (askOrderProbability <= prng.nextDouble()) {
+      MarketAskOrder(sender, quantity, timestamp(), tradable, uuid())
+    } else {
+      MarketBidOrder(sender, quantity, timestamp(), tradable, uuid())
+    }
+  }
+
+  /** Generates a collection of random market orders for some tradable.
+    *
+    * @param numberOrders
+    * @param tradable
+    * @return
+    */
+  protected def generateMarketOrders(numberOrders: Int, tradable: Tradable) = {
+    for (i <- 1 to numberOrders) yield generateMarketOrder(tradable)
+  }
+
+  /** Generates a random order for some tradable.
+    *
+    * @param tradable
+    * @return
+    */
+  protected def generateOrder(tradable: Tradable): Order = {
+    if (limitOrderProbability <= prng.nextDouble()) {
+      generateLimitOrder(tradable)
+    } else {
+      generateMarketOrder(tradable)
+    }
+  }
+
+  /** Generates a collection of random orders for some tradable.
+    *
+    * @param numberOrders
+    * @param tradable
+    * @return
+    */
+  protected def generateOrders(numberOrders: Int, tradable: Tradable) = {
+    for (i <- 1 to numberOrders) yield generateOrder(tradable)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use braces in for comprehensions](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/2385917881/issues/source?bid=2921361&fileBranchId=3228019#l123)